### PR TITLE
Fix build on Haiku beta alternative

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

This is an alternative code change related to #17857 

The implementation is different, by adding `-fno-stack-protector` to `ccflags` in hint files (currently `amigaos.sh` and `haiku.sh` but if any other platform needs it, no `Configure` change will be required then) and it inhibits the add of `-fstack-protector*` family in `Configure` script.

It still contains Haiku fixes (errno fix and location changes).

Best regards.

Thibault

### CODE REVIEW

**Comment by Leont** [comment]
> The implementation is different, by adding -fno-stack-protector to ccflags in hint files (currently amigaos.sh and haiku.sh but if any other platform needs it, no Configure change will be required then) and it inhibits the add of -fstack-protector* family in Configure script.

That does look like a scalable approach to me.

**Comment by thibaultduponchelle** [comment]
Rebased, it is ready to review and merge @khwilliamson

File: hints/amigaos.sh
Comment (atoomic) [inline hints/amigaos.sh:10]: I see `amigaos` was removed from Configure, were you able to test this change?

Comment (thibaultduponchelle) [reply hints/amigaos.sh:10]: No

Comment (thibaultduponchelle) [reply hints/amigaos.sh:10]: If you prefer, #17857 is safer.

**Comment by atoomic** [comment]
Indeed #17857 is more scoped on the case subject
I now understand better the scope of this case, this looks good to me, but would wait for a second review

**Comment by khwilliamson** [comment]
@atoomic which are you saying looks good to you; this or #17857

**Comment by atoomic** [comment]
@khwilliamson #17860 looks fine to me

**Comment by thibaultduponchelle** [comment]
Hello,

Since we last discussed about this PR, I tried VERY hard to test this change on `amigaos`.
I spent literally weeks to learn amigaos 4.1 and try to compile perl 5.32.0 on it but finally I was unfortunately not able to run a full compilation. 

What I can say : 
* The Configure process was OK (but it's only a "syntax" test actually)
* There seems to be other things to do (signals, pipe, network) for porting perl 5.32 to amigaos 4.1, so in the theory if my change is breaking the amigaos 4.1 perl port, it is probably already broken for some other reasons (bad argument I know).

I contacted @broadblues that contributed perl 5.22 amigaos 4.1 port in the past (and several great fixes, really wow) and he helped me a lot. 

Then what can I say about this PR ?
* Either we want to take 0 risk with amigaos port then we reopen and merge my other PR that is less elegant but at least do not touch amigaos at all.
* Either we keep this PR open but want an amigaos 4.1 fully working port to be delivered along with it (only way to validate `amigaos 4.1`) and therefore we need to wait that I get some good Amiga hardware to do the perl 5.32 port. In the coming months, years, or never.
* Or we merge this PR as is with a small risk or technical debt to be handled when someone will work on perl 5.32 amigaos port

I'm not arguing to absolutely merge this PR, I would, if you let me decide alone :rofl: but honestly I want to have your opinion/decision to move forward.

Thank you a lot

**Comment by thibaultduponchelle** [comment]
What do you think @khwilliamson @atoomic ?

**Review by atoomic — APPROVED** [review]

**Comment by atoomic** [comment]
Thanks @thibaultduponchelle for the long explanation and research.
You are probably right about amiga os being broken and without access to a server and/or CI it's hard to know in which state we are.

I think we should move forward and merge this pull request.
@khwilliamson what's your opinion on it?

**Comment by Leont** [comment]
> Thanks @thibaultduponchelle for the long explanation and research.
> You are probably right about amiga os being broken and without access to a server and/or CI it's hard to know in which state we are.
> 
> I think we should move forward and merge this pull request.

I agree

**Comment by broadblues** [comment]
Hi,
Wish I'd been added to this thread earlier and sorry for replying after the fact. (House move) I didn't get from tibs emails why he was trying to build for AmigaOS (my bad) I now understand and can say that -fstack-protector* and -fno-stack-protector are all no ops on the AmigaOS gcc port, so this change will make no difference to amigaos builds of perl.

WRT to actually building perl on AmigaOS the hints file needs a minor tweak WRT to pipe and mosts seds are broken except my own port which seems not to have proliferated but that's off topic for here. :-)

**Comment by xsawyerx** [comment]
> Hi,
> Wish I'd been added to this thread earlier and sorry for replying after the fact. (House move) I didn't get from tibs emails why he was trying to build for AmigaOS (my bad) I now understand and can say that -fstack-protector* and -fno-stack-protector are all no ops on the AmigaOS gcc port, so this change will make no difference to amigaos builds of perl.

Thanks for the input.

> WRT to actually building perl on AmigaOS the hints file needs a minor tweak WRT to pipe and mosts seds are broken except my own port which seems not to have proliferated but that's off topic for here. :-)

Is there a corresponding ticket to this? It would be good to bring in your fixes.

**Pull request merged**
